### PR TITLE
Make queue manager configurable.

### LINKED ISSUE #2990: Make the queue manager config configurable
This has been asked for by two people on the mailing list (and one privately) - https://groups.google.com/forum/#!msg/prometheus-developers/sF74LorpNOE/YUI9eUXGCgAJ, reasoning being that the default values for MaxSamplesPerSend (100) and MaxShards (1000) are not optimal for all TSDBs.

### PR DESCRIPTION
**Pull request by tomwilkie**
Fixes #2990 

Also, invoke `HTTPClientConfig.validate` manually, as it won't be called automatically.

### CODE REVIEW

File: config/config.go [hunk 1]
```diff
@@ -1404,12 +1424,43 @@ func (c *RemoteWriteConfig) UnmarshalYAML(unmarshal func(interface{}) error) err
...
 		return err
 	}
 	return nil
 }
 
+// QueueManagerConfig is the configuration for the queue used to write to remote
+// storage.
+type QueueManagerConfig struct {
+	// Number of samples to buffer per shard before we start dropping them.
+	QueueCapacity int
```
Comment (brancz) [inline config/config.go]: These all need yaml tags.

File: config/config.go [hunk 2]
```diff
@@ -1391,7 +1410,8 @@ type RemoteWriteConfig struct {
 
 	// We cannot do proper Go type embedding below as the parser will then parse
 	// values arbitrarily into the overflow maps of further-down types.
-	HTTPClientConfig HTTPClientConfig `yaml:",inline"`
+	HTTPClientConfig   HTTPClientConfig   `yaml:",inline"`
+	QueueManagerConfig QueueManagerConfig `yaml:",inline"`
```
Comment (brancz) [inline config/config.go]: As we inline the `HTTPClientConfig` in other places I guess it's fine, but I think this config should not be. Wdyt @brian-brazil ?

File: config/config.go (hunk 1 above)
Comment (tomwilkie) [inline config/config.go]: Ah yes, good point.

File: config/config.go (hunk 2 above)
Comment (brian-brazil) [inline config/config.go]: I think this is a section which will get many more options in the future, and which may be extra-experimental so my instinct would be to not inline it.

Comment (tomwilkie) [inline config/config.go]: SGTM

**Comment by tomwilkie** [comment]
@brancz can I take the "looking good" as an LGTM?

**Comment by brancz** [comment]
yep, sorry for not following up. lgtm :+1:

**Pull request merged**